Tasks/PublishCodeCoverageResultsV2/task.json

```diff
@@ -0,0 +1,61 @@
+{
+    "id": "2A7EBC54-C13E-490E-81A5-D7561AB7CD97",
+    "name": "PublishCodeCoverageResults",
+    "friendlyName": "Publish code coverage results",
+    "description": "Publish any of the  code coverage results supported by report generator from a build",
+    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results",
+    "helpMarkDown": "[Learn more about this task](https://go.microsoft.com/fwlink/?LinkID=626485)",
+    "category": "Test",
+    "visibility": [
+        "Build"
+    ],
+    "runsOn": [
+        "Agent",
+        "DeploymentGroup"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 2,
+        "Minor": 158,
+        "Patch": 6
+    },
+    "demands": [],
+    "minimumAgentVersion": "2.102.0",
+    "instanceNameFormat": "Publish code coverage results",
+    "inputs": [
+        {
+            "name": "summaryFileLocation",
+            "type": "multiLine",
+            "label": "Summary file",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Paths to summary files containing code coverage statistics, such as line, method, and class coverage. The value may contain minimatch patterns. For example: `$(System.DefaultWorkingDirectory)/MyApp/**/site/cobertura/coverage.xml`. Supports multiple lines of minimatch patterns. [More information](https://aka.ms/minimatchexamples)"
+        },
+        {
+            "name": "pathToSources",
+            "type": "string",
+            "label": "Path to Source files",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Path to source files is required when coverage XML reports do not contain absolute path to source files. For e.g., JaCoCo reports do not use absolute paths and when publishing JaCoCo coverage for Java apps, the pattern would be similar to `$(System.DefaultWorkingDirectory)/MyApp/src/main/java/`.<br />This input is also needed if tests are run in a docker container. This input should point to absolute path to source files on the host. For e.g., `$(System.DefaultWorkingDirectory)/MyApp/`"
+        },
+        {
+            "name": "failIfCoverageEmpty",
+            "type": "boolean",
+            "label": "Fail when code coverage results are missing",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Fail the task if code coverage did not produce any results to publish."
+        }
+    ],
+    "execution": {
+        "Node": {
+            "target": "publishcodecoverageresults.js",
+            "argumentFormat": ""
+        }
+    },
+    "messages": {
+        "FoundNMatchesForPattern": "Found %s result(s) matching pattern: %s",
+        "NoCodeCoverage": "No code coverage results were found to publish."
+    }
+}
```

Comment:
I don't see the report files input - we need that for repos that don't get web-reporting.
And the checkbox to auto-generate versus upload HTML?

Reply:
I thought the report directory stuff and upload local html report was to be left with V1, with V2 we assumed auto generation